ecosystem/sep-0009.md

```diff
@@ -148,8 +148,34 @@ single multi-line string in the `address` field. This allows any address in the
 number of fields. If address parsing is necessary, parsing will be easier since the country, city, and postal code are
 already separate fields.
 
+## Card fields
+
+To pass card (such as credit card) details to the application, the client should pass card details via an object defined
+below. Note, that it's possible to either pass card details to the application, or the token, representing the card.
+This token may be fetched from a third-party source prior, for example, it can be a
+[Stripe token](https://stripe.com/docs/api/tokens/create_card), or any other service offering similar functionality may
+be used. When token is used, application should notify clients about the type of the token that it is expecting to
+receive. Usually, application would require either `token`, or set of: `number`, `expiration_date`, `cvc` and
+`holder_name` to be provided, but some applications may require extra fields.
+
+| Name                     | Type   | Description                                                                                 |
+|--------------------------| ------ | ------------------------------------------------------------------------------------------- |
+| `card.number`            | number | Card number                                                                                 |
+| `card.expiration_date`   | date   | Expiration month and year (e.g. `11-29`, November 2029)                                     |
+| `card.cvc`               | number | CVC number (3 digits on the back of the card)                                               |
+| `card.holder_name`       | string | Name of the card holder                                                                     |
+| `card.type`              | string | Type of the bank card used (e.g. debit, credit)                                             |
```

Comment:
Should we define a fixed set of values here or is that not feasible? Is it strictly necessary too, or can the type of card be inferred based on other attributes like the number?

Reply:
This field is optional, I believe you can technically deduce type based on the number, but some (very few) merchants ask to specify type implicitly (e.g. if they don't support specific combination, like visa credit, so they can do validation on the FE, I believe)
I decided to add it for this reason, but I don't mind removing it

Reply:
Lets remove it until we get requests to add it. I'd rather not have a field that could hold arbitrary values. Enums are better.